File: programmers/level2/가장 큰 정사각형 찾기.py
def solution(board):
    for y in range(1, len(board)):
        for x in range(1, len(board[0])):
            if board[y][x] >= 1:
                board[y][x] = min(board[y][x-1],board[y-1][x],board[y-1][x-1])+1
    return max([num for row in board for num in  row])**2

File: programmers/level2/가장 큰 정사각형 찾기.java
class Solution {
    public int solution(int [][]board) {
        int maximum = board[0][0];
        for(int i=1; i<board.length; i++) {
            for(int j=1; j<board[0].length; j++) {
                if(board[i][j]>=1) {
                    board[i][j] = Math.min(board[i][j-1], Math.min(board[i-1][j], board[i-1][j-1])) + 1;
                    maximum = Math.max(maximum, board[i][j]);
                }
            }
        }
        return maximum*maximum;
    }
}